### DISCUSSION TITLE
I wish it were easier to add a one-off custom rule within a project

### DISCUSSION THREAD

**Discussion by jedwards1211** [eslint/eslint#17234] [General]
My company wants to implement a basic todo linting rule for our monorepo with slightly different logic than any plugins that are out there... I imagined I would just simply add a custom rule JS file somewhere in the monorepo and then refer to it in eslint config, but it's not that simple.

`--rulesdir` is deprecated and suggests using custom plugins instead.  It's not worth spending the time on making a separate reusable package at the moment, so I still wanted to put the plugin module within the repo.  But `eslint` doesn't accept `require.resolve` paths to plugins or a `./` relative path from the repo root.  Some guides recommend using `link:./path/to/plugin` to install the plugin into `node_modules`, but `yarn` copies the folder into `node_modules` instead of symlinking it.  If I could use `pnpm` I would, but that would be a big change for this company monorepo.

So I don't have any good options for writing up a lightweight custom rule for this project only.  I guess I'll just use the deprecated `--rulesdir` but I wish I had an option that feels more long term.  I'm sure you have reasons for wanting people to only make shareable plugins and configs, but it's frustrating that I don't have any good way to quickly accomplish my work objectives...is there any way I can do this?

**Comment by jedwards1211** [comment]
The docs for custom rules also say:

> The thing that makes ESLint different from other linters is the ability to define custom rules at runtime. This is perfect for rules that are specific to your project or company and wouldn’t make sense for ESLint to ship with or be included in a plugin. Just write your rules and include them at runtime.

If this is the case then why is `--rulesdir` deprecated?  Did this use case stop being a priority?

**Comment by jedwards1211** [comment]
Actually `--rulesdir` isn't really usable for me either since it seems to be CLI-only option.

**Comment by jedwards1211** [comment]
Okay I was wrong about `link:./path/to/plugin`, yarn seems to create a hardlink, so I can at least work with that

**Comment by mdjermanovic** [comment]
@jedwards1211 sorry for the confusion! We support both sharing rules in plugins and using local in-project rules. We're aware that the current solutions for using custom rules without publishing a plugin are not ideal. The [new config system](https://eslint.org/docs/latest/use/configure/configuration-files-new) will make it much easier to use local rules:

https://eslint.org/blog/2022/08/new-config-system-part-2/#from---rulesdir-to-runtime-plugins

> The docs for custom rules also say:
> 
> > The thing that makes ESLint different from other linters is the ability to define custom rules at runtime. This is perfect for rules that are specific to your project or company and wouldn’t make sense for ESLint to ship with or be included in a plugin. Just write your rules and include them at runtime.
> 
> If this is the case then why is `--rulesdir` deprecated? Did this use case stop being a priority?

This use case is still a priority. `--rulesdir` is marked deprecated because this option will be removed once the new config system replaces the old one, so we're not making any further changes or additions to this particular functionality. 

> `--rulesdir` is deprecated and suggests using custom plugins instead.

I can see how this can lead to the conclusion that we want people to only make shareable plugins, which isn't the case. We should improve the documentation to clarify why is `--rulesdir` deprecated.

**Reply by jedwards1211** [reply]
Oh nice, that all sounds great!  Thanks for letting me know!
